### PR TITLE
[SYCL][AMD] Propagate metadata in createURProgram

### LINKED ISSUE #14841: Basic/reqd_work_group_size.cpp

### PR DESCRIPTION
**Pull request by Naghasan**
SYCL properties weren't converted when calling creatreURProgram, leading to issue in finalization during KernelFusion for AMD.

Fixes #14841

### CODE REVIEW

**Review by aarongreig — APPROVED** [review]

File: sycl/test-e2e/KernelFusion/lit.local.cfg [hunk 1]
```diff
@@ -2,7 +2,7 @@ import platform
 
 config.required_features += ['fusion']
 # TODO: Reenable hip, see https://github.com/intel/llvm/issues/14598
```
Comment (aelovikov-intel) [inline sycl/test-e2e/KernelFusion/lit.local.cfg:4]: Shouldn't this comment be dropped? Also, the description might make use of something like "Closes <issue-link>"

**Comment by Naghasan** [comment]
address a regression introduced with moving to UR https://github.com/intel/llvm/issues/14598

**Comment by aelovikov-intel** [comment]
> address a regression introduced with moving to UR #14598

What I meant is https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

File: sycl/test-e2e/KernelFusion/lit.local.cfg (hunk 1 above)
Comment (Naghasan) [reply sycl/test-e2e/KernelFusion/lit.local.cfg]: yes, fixed, thanks

**Review by aelovikov-intel — APPROVED** [review]

**Comment by Naghasan** [comment]
> > address a regression introduced with moving to UR #14598
> 
> What I meant is https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

That wasn't an answer to your comment. But doesn't apply here.

**Comment by sommerlukas** [comment]
> > address a regression introduced with moving to UR #14598
> 
> What I meant is https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

#14598 captures multiple issues that were detected during the PI removal. Not all of them are addressed by this PR, so I don't think it makes sense to use a keyword such as `closes`.

**Comment by sommerlukas** [comment]
Just remembered that this PR should also reactivate this test: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/NewOffloadDriver/diamond_shape.cpp

**Review by sommerlukas — APPROVED** [review]

**Comment by Naghasan** [comment]
Windows job failing for an unrelated reason

```
Post job cleanup.
Cleaning D:\github\_work\llvm\llvm/*
rm: cannot remove 'D:\github\_work\llvm\llvm/build-ee/ESIMD': Device or resource busy
```

fyi @intel/dpcpp-devops-reviewers

**Comment by sarnex** [comment]
I restarted the runner ~1hr ago, should be fixed (tm). I'll restart the job here.

**Comment by callumfare** [comment]
The DeviceGlobal tests that were recently disabled are fixed by this change and can be reenabled:
```
DeviceGlobal/device_global_arrow.cpp
DeviceGlobal/device_global_device_only.cpp
DeviceGlobal/device_global_operator_passthrough.cpp
DeviceGlobal/device_global_subscript.cpp
```

**Comment by Naghasan** [comment]
reenabled

**Comment by Naghasan** [comment]
ping @dpcpp-tools-reviewers

**Review by AlexeySachkov — APPROVED** [review]

**Comment by Naghasan** [comment]
@intel/llvm-gatekeepers the PR is ready to be merged

**Pull request merged**